--- a/xrt/src/runtime_src/core/common/detail/linux/trace.h
+++ b/xrt/src/runtime_src/core/common/detail/linux/trace.h
@@ -4,7 +4,10 @@
 
 #ifdef XRT_ENABLE_USDT
 # define SDT_USE_VARIADIC
-# include <sys/sdt.h>
+#define STAP_PROBEV(...)
+#define DTRACE_PROBE(...)
+#define DTRACE_PROBE1(...)
+#define DTRACE_PROBE2(...)
 #else
 # define STAP_PROBEV(provider, name, ...) do {} while (0)
 # define DTRACE_PROBE(provider, name)      do {} while (0)
